### PR TITLE
added yaml analog of config.toml

### PR DESCRIPTION
**Pull request by dylanbannon**
## what
* Adding `config.yaml` YAML analog of current TOML Hugo config file `config.toml`.

## why
* Standardize cloudposse's configurations using YAML

## references
* https://cloudposse.atlassian.net/browse/INTERNAL-1184

### CODE REVIEW

**Review by osterman — CHANGES_REQUESTED** [review]
Update all `Makefile` to properly parse or work with `yaml`

Note, you can use `yq` instead of `sed`

**Comment by osterman** [comment]
- https://github.com/cloudposse/docs/blob/77fec1fcd37a463960f301a9a6914945b52864d2/Makefile#L97
- https://github.com/cloudposse/docs/blob/master/Makefile#L99-L112

**Comment by osterman** [comment]
https://github.com/cloudposse/docs/blob/04f1d84eaaecb46a793eda9c5024af84ad50a389/.gitignore

**Review by osterman — APPROVED** [review]

**Comment by osterman** [comment]
@dylanbannon the `pip` version of `yq` is not what we standardize on. Generally we poopoo on non-binaries as we're not a python shop. Let's get it working on the `yq` that we use.

**Review by osterman — CHANGES_REQUESTED** [review]
Use the `yq` from `cloudposse/packages`

**Comment by osterman** [comment]
https://github.com/mikefarah/yq

**Comment by dylanbannon** [comment]
@osterman Just to clarify, do you want me to install `yq` using `cloudposse/packages` (i.e.,
```
curl -1sLf \
  'https://dl.cloudsmith.io/public/cloudposse/packages/setup.deb.sh' \
  | sudo -E bash
```)
or do you want me to use the Debian repos, like you did in the picture above?

File: Makefile [hunk 1]
```diff
@@ -94,21 +94,21 @@ test:
...
 release:
-	@[ "$(HUGO_CONFIG)" != "config.toml" ] || (echo "Cannot release with $(HUGO_CONFIG)"; exit 1)
+	@[ "$(HUGO_CONFIG)" != "config.yaml" ] || (echo "Cannot release with $(HUGO_CONFIG)"; exit 1)
 	@[ "$(HTMLTEST_CONFIG)" != ".htmltest.yml" ] || (echo "Cannot release with $(HTMLTEST_CONFIG)"; exit 1)
-	cat config.toml | \
-		sed 's,^baseURL.*,baseURL = "$(HUGO_URL)",' | \
-		sed 's,^publishDir.*,publishDir = "$(HUGO_PUBLISH_DIR)",' | \
-		sed 's,^editURL.*,editURL = "$(HUGO_EDIT_URL)",' \
-		> $(HUGO_CONFIG)
+	if [ "$(yq eval 'select(.editURL)' config.yaml)" ]; then \
```
Comment (osterman) [inline Makefile:104]: Note this is not run in a docker run

Comment (osterman) [reply Makefile:104]: Why do we need this conditional?

Comment (dylanbannon) [reply Makefile:104]: I hadn't looked atthe `config.toml` file closely enough and thought that `config.yaml` shuld have a top-level field called `editURL`, which it doesn't. In reality, the yaml field is `params.editURL` and it is there. So, long chain of reasoning, but none of it is really applicable, it turns out.

It's maybe worth noting, though,  that [the version yq we're using](https://mikefarah.gitbook.io/yq/) doesn't have any `if...else...` functionality that I could find, which is why I ended up implementing the if,else in bash.

Comment (dylanbannon) [reply Makefile:104]: (I simplified this in the most recent commit.)

**Review by osterman — APPROVED** [review]

**Pull request merged**